htdocs/product/class/product.class.php

```diff
@@ -2437,7 +2491,7 @@ public function fetch($id = '', $ref = '', $ref_ext = '', $barcode = '', $ignore
 			} else {
 				$sql .= " p.pmp,";
 			}
-			$sql .= " p.datec, p.tms, p.import_key, p.entity, p.desiredstock, p.tobatch, p.batch_mask, p.fk_unit,";
+			$sql .= " p.datec, p.tms, p.import_key, p.entity, p.desiredstock, p.tobatch, p.sell_or_eat_by_mandatory, p.batch_mask, p.fk_unit,";
```

Comment:
Since version 15, all new features that need change into database structure must be solitted in 2: one PR for sql changes so database change can be validated, and then one PR for code that implement/use the database.

Reply:
And it's in a separate PR #21709 you merged